test/sql/function/timestamp/test_icu_dateadd.test

```diff
@@ -0,0 +1,189 @@
+00:00:00.000612	2021-12-01 13:54:48.124068+00
+-00:00:00.000485	2021-12-01 13:54:48.122971+00
+
+#  interval + timestamp
+query II
+SELECT iv, iv + '2021-12-01 13:54:48.123456'::TIMESTAMPTZ FROM intervals;
+----
+1 year	2022-12-01 13:54:48.123456+00
+-1 years	2020-12-01 13:54:48.123456+00
+1 month	2022-01-01 13:54:48.123456+00
+-1 months	2021-11-01 12:54:48.123456+00
+1 year 1 month	2023-01-01 13:54:48.123456+00
+-1 years -3 months	2020-09-01 12:54:48.123456+00
+1 day	2021-12-02 13:54:48.123456+00
+-1 days	2021-11-30 13:54:48.123456+00
+32 days	2022-01-02 13:54:48.123456+00
+-40 days	2021-10-22 12:54:48.123456+00
+01:00:00	2021-12-01 14:54:48.123456+00
+-01:00:00	2021-12-01 12:54:48.123456+00
+11:00:00	2021-12-02 00:54:48.123456+00
+-14:00:00	2021-11-30 23:54:48.123456+00
+00:01:00	2021-12-01 13:55:48.123456+00
+-00:01:00	2021-12-01 13:53:48.123456+00
+00:06:00	2021-12-01 14:00:48.123456+00
+-01:12:00	2021-12-01 12:42:48.123456+00
+00:00:01	2021-12-01 13:54:49.123456+00
+-00:00:01	2021-12-01 13:54:47.123456+00
+00:00:23	2021-12-01 13:55:11.123456+00
+-00:01:58	2021-12-01 13:52:50.123456+00
+00:00:00.001	2021-12-01 13:54:48.124456+00
+-00:00:00.001	2021-12-01 13:54:48.122456+00
+00:00:00.91	2021-12-01 13:54:49.033456+00
+-00:00:00.15	2021-12-01 13:54:47.973456+00
+00:00:00.000001	2021-12-01 13:54:48.123457+00
+-00:00:00.000001	2021-12-01 13:54:48.123455+00
+00:00:00.000612	2021-12-01 13:54:48.124068+00
+-00:00:00.000485	2021-12-01 13:54:48.122971+00
+
+# timestamp - interval
+query II
+SELECT iv, '2021-12-01 13:54:48.123456'::TIMESTAMPTZ - iv FROM intervals;
+----
+1 year	2020-12-01 13:54:48.123456+00
+-1 years	2022-12-01 13:54:48.123456+00
+1 month	2021-11-01 12:54:48.123456+00
+-1 months	2022-01-01 13:54:48.123456+00
+1 year 1 month	2020-11-01 13:54:48.123456+00
+-1 years -3 months	2023-03-01 13:54:48.123456+00
+1 day	2021-11-30 13:54:48.123456+00
+-1 days	2021-12-02 13:54:48.123456+00
+32 days	2021-10-30 12:54:48.123456+00
+-40 days	2022-01-10 13:54:48.123456+00
+01:00:00	2021-12-01 12:54:48.123456+00
+-01:00:00	2021-12-01 14:54:48.123456+00
+11:00:00	2021-12-01 02:54:48.123456+00
+-14:00:00	2021-12-02 03:54:48.123456+00
+00:01:00	2021-12-01 13:53:48.123456+00
+-00:01:00	2021-12-01 13:55:48.123456+00
+00:06:00	2021-12-01 13:48:48.123456+00
+-01:12:00	2021-12-01 15:06:48.123456+00
+00:00:01	2021-12-01 13:54:47.123456+00
+-00:00:01	2021-12-01 13:54:49.123456+00
+00:00:23	2021-12-01 13:54:25.123456+00
+-00:01:58	2021-12-01 13:56:46.123456+00
+00:00:00.001	2021-12-01 13:54:48.122456+00
+-00:00:00.001	2021-12-01 13:54:48.124456+00
+00:00:00.91	2021-12-01 13:54:47.213456+00
+-00:00:00.15	2021-12-01 13:54:48.273456+00
+00:00:00.000001	2021-12-01 13:54:48.123455+00
+-00:00:00.000001	2021-12-01 13:54:48.123457+00
+00:00:00.000612	2021-12-01 13:54:48.122844+00
+-00:00:00.000485	2021-12-01 13:54:48.123941+00
+
+# Before the epoch
+query II
+SELECT iv, '1962-07-31 12:20:48.123456'::TIMESTAMPTZ + iv FROM intervals;
```

Comment:
Could we add some overflow tests with very large timestamps (see e.g. `test/sql/types/timestamp/timestamp_limits.test` for examples of large timestamps)?

Reply:
The limits testing for ICU dates is a bit messed up because they have different accuracies at those points. My goal with the new tests is to a) make sure nothing explodes and b) document the inconsistencies. Unless someone is doing ms accuracy analytics on servers deployed shortly after the development of speech, this should be good enough.